src/autogluon/bench/frameworks/multimodal/exec.py

```diff
@@ -13,12 +14,24 @@
     _TEXT_SIMILARITY,
 )
 from autogluon.bench.datasets.dataset_registry import multimodal_dataset_registry
-from autogluon.bench.utils.general_utils import NumpyEncoder
 from autogluon.multimodal import MultiModalPredictor
+from autogluon.multimodal import __version__ as ag_version
 
 logger = logging.getLogger(__name__)
 
 
+def _flatten_dict(dd, separator="_", prefix=""):
+    return (
+        {
+            f"{prefix}{separator}{k}" if prefix else k: v
+            for kk, vv in dd.items()
+            for k, v in _flatten_dict(vv, separator, kk).items()
+        }
+        if isinstance(dd, dict)
+        else {prefix: dd}
+    )
```

Comment:
I had to read this a few times to make sure I understood how it worked. Does this accomplish the same thing, and do you think its more readable?

```python
def _flatten_dict(data, separator='_', prefix=''):
    flattened = {}
    for key, value in data.items():
        if isinstance(value, dict):
            flattened.update(_flatten_dict(value, separator, prefix + key + separator))
        else:
            flattened[prefix + key] = value
    return flattened
```

Reply:
I like your refactor and I think they do the same thing. Will update!